dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1444,5 +1444,82 @@ def test_null_row_stats_correct_after_updates(self, *mocks):
         self.assertSetEqual({8}, profile._profile[1].null_types_index['None'])
 
 
+class TestProfilerFactoryClass(unittest.TestCase):
+
+    def test_profiler_factory_class_bad_input(self):
+        with self.assertRaisesRegex(ValueError, "Must specify 'profiler_type' "
+                                                "to be 'structured' or "
+                                                "'unstructured'."):
+            Profiler(pd.DataFrame([]), profiler_type="whoops")
+
+        with self.assertRaisesRegex(ValueError, "Data must either be imported "
+                                                "using the data_readers, "
+                                                "pd.Series, or pd.DataFrame."):
+            Profiler("whoops")
+
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    @mock.patch('dataprofiler.profilers.profile_builder.UnstructuredCompiler')
+    def test_profiler_factory_class_creates_correct_profiler(self, *mocks):
```

Comment:
should add a test to see what happens if we try to profile a series of dictionaries:
e.g.
```python
data = pd.Series([{'a': 1}, {'b': 2}, {'c': 3}])
```

Reply:
What is even the correct behavior in this case?

Reply:
it would be treated like the string case. Event hough i'd  prefer it to be unstructured, this case will likely be treated as structured.